components/script/dom/gpudevice.rs

```diff
@@ -214,6 +220,139 @@ impl GPUDevice {
+                                        },
+                                        blend: state.blend.as_ref().map(|blend| wgt::BlendState {
+                                            color: convert_blend_component(&blend.color),
+                                            alpha: convert_blend_component(&blend.alpha),
+                                        }),
+                                    })
+                                })
+                                .collect::<Vec<_>>(),
+                        ),
+                    }),
+                primitive: convert_primitive_state(&descriptor.primitive),
+                depth_stencil: descriptor.depthStencil.as_ref().map(|dss_desc| {
+                    wgt::DepthStencilState {
+                        format: convert_texture_format(dss_desc.format),
+                        depth_write_enabled: dss_desc.depthWriteEnabled,
+                        depth_compare: convert_compare_function(dss_desc.depthCompare),
+                        stencil: wgt::StencilState {
+                            front: wgt::StencilFaceState {
+                                compare: convert_compare_function(dss_desc.stencilFront.compare),
+                                fail_op: convert_stencil_op(dss_desc.stencilFront.failOp),
+                                depth_fail_op: convert_stencil_op(
+                                    dss_desc.stencilFront.depthFailOp,
+                                ),
+                                pass_op: convert_stencil_op(dss_desc.stencilFront.passOp),
+                            },
+                            back: wgt::StencilFaceState {
+                                compare: convert_compare_function(dss_desc.stencilBack.compare),
+                                fail_op: convert_stencil_op(dss_desc.stencilBack.failOp),
+                                depth_fail_op: convert_stencil_op(dss_desc.stencilBack.depthFailOp),
+                                pass_op: convert_stencil_op(dss_desc.stencilBack.passOp),
+                            },
+                            read_mask: dss_desc.stencilReadMask,
+                            write_mask: dss_desc.stencilWriteMask,
+                        },
+                        bias: wgt::DepthBiasState {
+                            constant: dss_desc.depthBias,
+                            slope_scale: *dss_desc.depthBiasSlopeScale,
+                            clamp: *dss_desc.depthBiasClamp,
+                        },
+                    }
+                }),
+                multisample: wgt::MultisampleState {
+                    count: descriptor.multisample.count,
+                    mask: descriptor.multisample.mask as u64,
+                    alpha_to_coverage_enabled: descriptor.multisample.alphaToCoverageEnabled,
+                },
+                multiview: None,
+            })
+        } else {
+            /*self.dispatch_error(webgpu::Error::Validation(String::from(
```

Comment:
Can we either convert this into a clearer TODO comment or remove the commented code?